types/time_test.go

```diff
@@ -56,17 +56,19 @@ func TestUnmarshalJSON(t *testing.T) {
 
 	require.ErrorIs(t, json.Unmarshal([]byte(`"blurp"`), &testTime), strconv.ErrSyntax)
 	require.Error(t, json.Unmarshal([]byte(`"123456"`), &testTime))
+
+	// Captures bad syntax when type should be time.Time (RFC3339)
+	require.ErrorIs(t, json.Unmarshal([]byte(`"2025-03-28T08:00:00Z"`), &testTime), strconv.ErrSyntax)
+	require.Error(t, json.Unmarshal([]byte(`"123456"`), &testTime))
```

Comment:
```suggestion
	require.Error(t, json.Unmarshal([]byte(`123456`), &testTime))
```
Otherwise this is a duplicated test from above